spyder/plugins/ipythonconsole/widgets/shell.py

```diff
@@ -484,16 +484,31 @@ def get_editorstack(self):
             return editor.get_current_editorstack()
         raise RuntimeError('No editorstack found.')
 
-    def handle_save_files(self):
-        """Save the open files."""
-        self.get_editorstack().save()
+    def _handle_get_file_code(self, filename):
+        """
+        Return the bytes that compose the file.
+
+        Bytes are returned instead of str to support non utf-8 files.
+        """
+        editorstack = self.get_editorstack()
+        if CONF.get('editor', 'save_all_before_run', True):
+            editorstack.save_all()
+        editor = self.get_editor(filename)
+
+        if editor is None:
+            # Load it from file instead
+            text, _enc = encoding.read(filename)
```

Comment:
I don't agree with this change, but in any case it's not necessary for this PR. So for now please leave the kernel to handle the case of running a file not present in the editor.

We can discuss later the merits of this change (I already read what you posted in PR #10115, but I'm still not sure about it).

Reply:
I will do that.

My reasoning was that I didn't want to have an inconsistancy in the source of the files:
 - `%run` runs files on the kernel machine.
 - `runfile` runs files on the frontend machine.

If I remove this change, `runfile` runs file on the frontend machine if they are open, otherwise on the kernel machine.